video/out/wayland_common.c

```diff
@@ -3510,8 +3491,19 @@ static void set_color_management(struct vo_wayland_state *wl)
         wp_image_description_creator_params_v1_set_max_cll(image_creator_params, lrintf(hdr.max_cll));
         wp_image_description_creator_params_v1_set_max_fall(image_creator_params, lrintf(hdr.max_fall));
     }
+
     struct wp_image_description_v1 *image_description = wp_image_description_creator_params_v1_create(image_creator_params);
+    struct wl_event_queue *cm_queue = wl_display_create_queue(wl->display);
+    wl_proxy_set_queue((struct wl_proxy *)image_description, cm_queue);
+    wl->image_desc_done = false;
     wp_image_description_v1_add_listener(image_description, &image_description_listener, wl);
+
+    /* Block here to ensure the compositor is ready to receive the
```

Comment:
Can't we instead of blocking thread, react in the callback and do rest of the work when compositor is ready for it?

Reply:
If you do a roundtrip and the event has not yet arrived, the process will be aborted when the event arrives after you've destroyed the queue.

Reply:
We could keep the queue around then